web/src/engine/package-cache/src/keyboardRequisitioner.ts

```diff
@@ -97,6 +97,18 @@ export default class KeyboardRequisitioner {
     this.pathConfig = pathConfig;
     this.cache = new StubAndKeyboardCache(keyboardLoader);
     this.cloudQueryEngine = new CloudQueryEngine(keyboardRequester, this.pathConfig);
+
+    // Handles keymanweb.com's precached keyboard array.  There is no associated promise,
+    // so there's nothing handling the `register` call's results otherwise.
+    this.cloudQueryEngine.on('unboundregister', (registration) => {
+      // Internal, undocumented use-case of `keyman.register`:  precached keyboard loading
+      // Other uses may trigger errors, especially if there's a type-structure mismatch.
+      // Those errors should not be handled here; let them surface.
+      if(Array.isArray(registration)) {
+        registration.forEach((entry) => {
+          this.cache.addStub(entry);
+        });
+    });
```

Comment:
Given `keyman.register` was completely undocumented (which is really not a good thing given it definitely is an internal API), it's a bit of a stretch to say this was an undocumented use-case: all use-cases were undocumented. 

It's perhaps something you did not consider when you changed the implementation to use promises. This issue underlines just how critical it is to avoid changing APIs, even internal undocumented ones, without full audit of _all_ uses.

Reply:
Yeah.  It _does_ help to know such uses exist, though - took a bit to connect the dots once I realized keymanweb.com's caching might be related to the reported issue, especially since nothing else related was broken.

Reply:
That's the _audit_ bit -- no one is going to remember everything, so code search is our friend here. For web, it's particularly troublesome given the websites can also interact. And 'register' is an unhelpfully common term so a search for this would have been particularly painful!

But in any case, searching api.keyman.com, keymanweb.com, help.keyman.com, and keyman.com sources any time any 'public' interface is touched on KeymanWeb will probably help with this in future. (All those sites use KeymanWeb or interact with KeymanWeb in some way or other, e.g. bookmarklet, embedded documentation OSK, etc.)